### DISCUSSION TITLE
Timefold for kitchen management

### DISCUSSION THREAD

**Discussion by NikitaObukhov** [TimefoldAI/timefold-solver#441] [General]
Hello everyone

I am new to Timefold AI, but it sees it can solve my problem. I need to write a program for restaurant kitchen management. The goal is to calculate order completeness time based on kitchen workload (equipment like ovens and staff, number of orders, and meals in it)

I was looking at or-tools Job shop problem and it seems suitable but very low-level. Timefold provides some useful abstractions and seems to be easier to work with. But I could not find some things here:
1. For a flexible job shop, different machines have different productivity. E.g. Task A can be completed at N seconds and Machine A, and in M seconds on machine B. I [found this page](https://timefold.ai/docs/timefold-solver/latest/use-cases-and-examples/project-job-scheduling/project-job-scheduling), but did not find any code example
2. Machines can be paused/unpaused during the shift time
3. Each machine may have it's own shift time
4. It is also unclear how to run it continuously, I mean the program must calculate new incoming orders and persist already calculated. It looks like job pinning and schedule repository suites this needs, It also has great docs "[Responding to changes](https://timefold.ai/docs/timefold-solver/latest/responding-to-change/responding-to-change)". But are there any examples with this thing running in real time and replanning when new jobs arrive?

Thanks in advance!

**Comment by triceo** [comment]
Hello @NikitaObukhov!

Thanks for the interest in Timefold Solver. What you are looking for is all technically possible, and we have bits and pieces to show. I recommend splitting the problem into two parts:

1. Modelling the domain and making sure you can solve it. For that, you can probably take some inspiration in our [project job scheduling example](https://github.com/TimefoldAI/timefold-solver/tree/main/examples/src/main/java/ai/timefold/solver/examples/projectjobscheduling), but you will still need to modify it for your needs.  
2. Taking this domain and turning it into a web app that can respond to events. For the realtime part, you can take a look at our [Call Center quickstart](https://github.com/TimefoldAI/timefold-quickstarts/tree/stable/use-cases/call-center).

As you can see, delivering an end-to-end solution like that is a considerable amount of work.
Maybe [one of our partners](https://timefold.ai/partners) could help you?

**Reply by NikitaObukhov** [reply]
Thank you so much! Well I am just a developer who was assigned to this task. If I could not solve it on my own we will probably look for some consulting
